### ISSUE TITLE
 protoc-gen-openapiv2 generate odd prefix for Location message definition

### ISSUE THREAD

**Issue by nazy** [grpc-ecosystem/grpc-gateway#2191]
## 🐛 Bug Report

When I generate Open API definition with  `protoc-gen-openapiv2`, the definition for `Location` message has an odd prefix.  
I believe the prefix of definitions is the last part of package name but only for `Location` message, it also contains the parent package name.  
e.g. If package name is `example.test`. Usual message definition's prefix is `test`, but only for `Location`, the prefix will be `exampletest`.

## To Reproduce

1. Setup `grpc-gateway` according to `README.md`'s `Installation` section
1. Clone `googleapis`
    ```sh
    $ git clone --depth 1 https://github.com/googleapis/googleapis.git
    ```
1. Create the following proto file
    ```proto
    syntax = "proto3";
    package example.test;
    option go_package = "example/test";
    import "google/api/annotations.proto";
    import "google/protobuf/empty.proto";
    
    service TestService {
      rpc Test(TestRequest) returns (google.protobuf.Empty) {
          option (google.api.http) = {};
      }
    }
    
    message TestRequest {
      Location x = 1;
      NotLocation y = 2;
    }
    message Location { }
    message NotLocation { }
    ```
1.  Generate OpenAPI definitions
    ```sh
    $ protoc -I googleapis -I . --openapiv2_out . test.proto 
    ``` 
1. Check `Location`'s definition
    ```sh
    $ grep -i location test.swagger.json
        "exampletestLocation": {
        "testNotLocation": {
    ```

## Expected behavior

All message definitions' prefix are the last part of the package name.

e.g. package is example.test
* Location -> testLocation
* AnyOther -> testAnyOther

## Actual Behavior

e.g. package is example.test
* Location -> exampletestLocation
* AnyOther -> testAnyOther

## Your Environment

* OS: Ubuntu 20.04 on Vagrant (box: generic/ubuntu2004).
* golang : go version go1.16.5 linux/amd64
* protoc: libprotoc 3.17.1
* protoc-gen-openapiv2: Version dev, commit unknown, built at unknown
    * but I believe it is v2.4.0
* go.mod
    ```
    module example.com/test
    
    go 1.16
    
    require (
            github.com/grpc-ecosystem/grpc-gateway/v2 v2.4.0
            google.golang.org/grpc/cmd/protoc-gen-go-grpc v1.1.0
            google.golang.org/protobuf v1.26.0
    )
    ```

**Comment by johanbrandhorst** [comment]
That's a little surprising, we have an option to turn on fully qualified names normally, they shouldn't be fully qualified by default. Could you compare your expected behavior against what your tests and examples do? I can't tell just from this brief description whether this is a bug or just unexpected behavior.

See https://github.com/grpc-ecosystem/grpc-gateway/blob/master/protoc-gen-openapiv2/main.go#L29

**Comment by nazy** [comment]
Thank you for your comment. Additional testing has been performed. The results are shown below.  
With `fqn_for_openapi_name=true`, I got an expected result.

## Brief

* Without `fqn_for_openapi_name=true`
    * "exampletestLocation": {
    * "testNotLocation": {
* With `fqn_for_openapi_name=true`
    * "example.test.Location": {
    * "example.test.NotLocation": {

## Detail

* protoc -I googleapis -I . --openapiv2_out . test.proto 
```json
{
  "** SNIP **": "",
  "definitions": {
    "exampletestLocation": {
      "type": "object"
    },
    "protobufAny": {
      "type": "object",
      "properties": {
        "typeUrl": {
          "type": "string"
        },
        "value": {
          "type": "string",
          "format": "byte"
        }
      }
    },
    "rpcStatus": {
      "type": "object",
      "properties": {
        "code": {
          "type": "integer",
          "format": "int32"
        },
        "message": {
          "type": "string"
        },
        "details": {
          "type": "array",
          "items": {
            "$ref": "#/definitions/protobufAny"
          }
        }
      }
    },
    "testNotLocation": {
      "type": "object"
    }
  }
}
```

* protoc -I googleapis -I . --openapiv2_out . --openapiv2_opt fqn_for_openapi_name=true test.proto 
```json
{
  "** SNIP **": "",
  "definitions": {
    "example.test.Location": {
      "type": "object"
    },
    "example.test.NotLocation": {
      "type": "object"
    },
    "google.protobuf.Any": {
      "type": "object",
      "properties": {
        "typeUrl": {
          "type": "string"
        },
        "value": {
          "type": "string",
          "format": "byte"
        }
      }
    },
    "google.rpc.Status": {
      "type": "object",
      "properties": {
        "code": {
          "type": "integer",
          "format": "int32"
        },
        "message": {
          "type": "string"
        },
        "details": {
          "type": "array",
          "items": {
            "$ref": "#/definitions/google.protobuf.Any"
          }
        }
      }
    }
  }
}
```

**Comment by johanbrandhorst** [comment]
Does this solve your problem? I'm not sure what you're asking for here.

**Comment by nazy** [comment]
My problem itself was solved with fqn_for_openapi_name=true. Thank you for your advice!

However, the issue itself is not solved. I just reported a strange behavior looks like a bug, so I just leave it open.  
Here is an issue tracker, and not a customer support center. Therefore I wrote something to reproduce the issue and support your investigation.  
This issue has a workaround and I think this issue is a corner case, therefore I don't care you'll close it with `Won't fix`.

**Comment by nazy** [comment]
In addition to, I did some investigation before submit this issue.
1. I've skimmed grpc-gateway source code, but I could not found the cause. At least 'Location' is not defined in the code.
2. I suspected a name conflict with other .proto, definitions but it didn't matter. In googleapis, there are several Messages named `Location`. And there is no Message named Location in googleapis/google/api, which I needed, so I just extracted and used that part, but the result was the same.

Also, I tried with v2.5.0 today, but the result was same.
